src/lib.rs

```diff
@@ -442,3 +442,24 @@ storage_types! {
         }
     }
 }
+
+/// Shared tools for parsing quantities from strings.
```

Comment:
`mod str` isn't just for parsing. `core::str`'s documentation is `String manipulation`, `std::str` is `Unicode string slices`. Consider changing to `Unicode string slice manipulation for quantities.`